### PR TITLE
TIED-118 SonarCloud issues

### PR DESCRIPTION
**Pull request by mikkojamG**
## Description

Fixing SonarCloud issues.
Bugs:
- https://sonarcloud.io/project/issues?open=AYgKD3QavNC-zMHTKAE5&id=City-of-Helsinki_helerm-ui
- https://sonarcloud.io/project/issues?open=AYgKD3RRvNC-zMHTKAFF&id=City-of-Helsinki_helerm-ui

Code Smells:
- Refactoring functions to reduce complexity
- Prefer using optional chain expression
- Replace `Object.prototype.hasOwnProperty.call` with `Object.hasOwn`
- Component prop validation inconsistencies
- Extract nested ternary operations
- Pass client to `useMemo` hook (ClientProvider)
- Replacing deprecated `createStore`
- `String.starsWith` instead of `charAt`

## Related Issue(s)

**[TIED-118](https://helsinkisolutionoffice.atlassian.net/browse/TIED-118)**

## Screenshots / Videos

[TIED-118]: https://helsinkisolutionoffice.atlassian.net/browse/TIED-118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/utils/validators.js [hunk 1]
```diff
@@ -31,50 +30,50 @@ export const validateConditionalRules = (key, attributeTypes, attributes) => {
...
+      type
+    );
+
+    return (isRequired && isRequiredInType && !objHasRuleAttribute) ||
+      (objHasRuleAttribute && !isValid && !allowValuesOutsideChoices) ||
+      (!isAttributeAllowedInType && objHasRuleAttribute)
+  })
+
+  const conditionallyRequired = keys
+    .filter(((key) => rules[key].requiredIf.length > 0))
```
Comment (arpargo) [inline src/utils/validators.js:59]: I believe you can leave out one set of brackets here.

**Review by arpargo — CHANGES_REQUESTED** [review]
Left one comment, couldn't find anything else and ui booted up fine locally and seemed to work as usual.

File: src/utils/validators.js (hunk 1 above)
Comment (mikkojamG) [reply src/utils/validators.js:59]: Whoops, will fix 👀

Comment (mikkojamG) [reply src/utils/validators.js]: https://github.com/City-of-Helsinki/helerm-ui/pull/493/commits/fbb6b21dc6c62a650329de64a01bb83219537b41

**Review by arpargo — APPROVED** [review]
LGTM

**Pull request merged**